pymongoexplain/__main__.py

```diff
@@ -14,47 +14,35 @@
 
 
 from pymongo import monitoring, MongoClient
-from pymongo.collection import Collection
+from pymongo.collection import Collection, ReturnDocument
+from pymongo.cursor import CursorType
+from pymongo.common import BaseObject
+from .explainable_collection import ExplainCollection
 import sys
 from bson.son import SON
 
-class CommandLogger(monitoring.CommandListener):
-    def __init__(self):
-        self.payloads = []
+#do some monkey patching to run explain before the actual command
+old_function_names = ["update_one", "replace_one", "update_many", "delete_one",
+                      "delete_many", "aggregate", "watch", "find", "find_one",
+                      "find_one_and_delete", "find_one_and_replace",
+                      "find_one_and_update", "count_documents",
+                      "estimated_document_count", "distinct"]
+old_functions = [getattr(Collection, i) for i in old_function_names]
 
-    def started(self, event):
-        print("started")
-        self.payloads.append(event.command)
 
-    def succeeded(self, event):
-        pass
+def make_func(old_func, old_func_name):
+    def new_func(self: Collection, *args, **kwargs):
+        print(getattr(ExplainCollection(self), old_func_name)(*args,
```

Comment:
What do you think about using the logging module instead of `print`? See https://docs.python.org/3/howto/logging.html#logging-basic-tutorial